OpenCover.UI.TestDiscoverer/NUnit/NUnitDiscoverer.cs

```diff
@@ -54,12 +54,36 @@ protected override List<TestClass> DiscoverTestsInAssembly(string dllPath, Assem
 
                 if (isNUnitTest)
                 {
+                    var traits = new List<string>();
+                    try
+                    {
+                        var currentType = type;
+                        while (true)
+                        {
+                            foreach (var attribute in currentType.CustomAttributes)
+                            {
+                                AddTraits(traits, attribute, typeof(CategoryAttribute));
+                            }
+
+                            if (currentType.BaseType != null)
+                            {
+                                currentType = currentType.BaseType.Resolve();
+                            }
+                            else
+                            {
+                                break;
+                            }
+                        }
+                    }
+                    catch { }
+
                     var TestClass = new TestClass
                     {
                         DLLPath = dllPath,
                         Name = type.Name,
                         Namespace = type.Namespace,
-                        TestType = TestType.NUnit
+                        TestType = TestType.NUnit,
+                        Traits = traits.Count > 0 ? traits.ToArray() : null
```

Comment:
I think it's ok to just do
`Traits = traits.ToArray()`
here. It makes sure Traits is never null and it's no problem to have an empty array in there